### PR TITLE
Extend interpretable embedding with multiple args

### PR DESCRIPTION
**Pull request by NarineK**
This PR extends InterpretableEmbeddingBase to be more generic and support nested embeddings of any types and number of inputs.
I've also added test cases and documentation related to this PR.

In addition to that it looks like in the IMDB tutorial the device was missing and some minor fixes. Adding that as well.

### CODE REVIEW

File: captum/attr/_models/base.py [hunk 1]
```diff
@@ -5,10 +5,10 @@
 import warnings
 
 from functools import reduce
-from torch.nn import Embedding
+from torch.nn import Module
 
 
-class InterpretableEmbeddingBase(Embedding):
+class InterpretableEmbeddingBase(Module):
```
Comment (orionr) [inline captum/attr/_models/base.py]: Should this now be `InterpretableModuleBase` or is it still only Embedding even though you are inheriting from Module?

File: captum/attr/_models/base.py [hunk 2]
```diff
@@ -21,20 +21,34 @@ class InterpretableEmbeddingBase(Embedding):
     """
 
     def __init__(self, embedding, full_name):
-        super().__init__(embedding.num_embeddings, embedding.embedding_dim)
+        Module.__init__(self)
+        self.num_embeddings = (
+            embedding.num_embeddings if hasattr(embedding, "num_embeddings") else None
+        )
+        self.embedding_dim = (
+            embedding.embedding_dim if hasattr(embedding, "embedding_dim") else None
```
Comment (orionr) [inline captum/attr/_models/base.py]: Can you just use `getattr` for both of these with the None default?

File: tutorials/IMDB_TorchText_Interpret.ipynb [hunk 3]
```diff
@@ -15,14 +15,14 @@
    "source": [
     "This notebook loads pretrained CNN model for sentiment analysis on IMDB dataset. It makes predictions on test samples and interprets those predictions using integrated gradients method.\n",
     "\n",
-    "The model was trained using an open source sentiment analysis tutorials described in: https://github.com/bentrevett/pytorch-sentiment-analysis/blob/master/4%20-%20Convolutional%20Sentiment%20Analysis.ipynb\n  ",
-    "\n  ",
```
Comment (orionr) [inline tutorials/IMDB_TorchText_Interpret.ipynb]: Doesn't seem  like we'd want these changes?

File: tutorials/IMDB_TorchText_Interpret.ipynb [hunk 4]
```diff
@@ -15,14 +15,14 @@
...
+    "The model was trained using an open source sentiment analysis tutorials described in:https://github.com/bentrevett/pytorch-sentiment-analysis/blob/master/4%20-%20Convolutional%20Sentiment%20Analysis.ipynb\n",
+    "  \n",
     "**Note:** Before running this tutorial, please install the spacy package, and its NLP modules for English language."
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": 1,
+   "execution_count": 24,
```
Comment (orionr) [inline tutorials/IMDB_TorchText_Interpret.ipynb]: Same? Don't want these?

File: tutorials/IMDB_TorchText_Interpret.ipynb [hunk 5]
```diff
@@ -467,7 +467,7 @@
   },
   {
    "cell_type": "code",
-   "execution_count": 18,
+   "execution_count": 20,
```
Comment (orionr) [inline tutorials/IMDB_TorchText_Interpret.ipynb]: Maybe just run straight through and save that?

File: captum/attr/_models/base.py (hunk 1 above)
Comment (orionr) [reply captum/attr/_models/base.py]: Actually, what's the specific motivation for inheriting differently? Seems like you'd want to still just have Embedding per the code. If you still need this please add a good comment about why.

Comment (NarineK) [reply captum/attr/_models/base.py]: Good point! In a general case Embedding doesn't have to extend `nn.Embedding` but an `nn.Module`. An example of it is `BertEmbedding`:https://github.com/huggingface/transformers/blob/master/transformers/modeling_bert.py#L151

File: captum/attr/_models/base.py (hunk 2 above)
Comment (NarineK) [reply captum/attr/_models/base.py]: Good point! Fixed!

**Pull request closed without merging**